example-app/src/main/java/backtraceio/backtraceio/CustomRequestHandler.java

```diff
@@ -0,0 +1,165 @@
+import backtraceio.library.common.URLRequestHandler;
+import backtraceio.library.models.BacktraceData;
+import backtraceio.library.models.BacktraceNativeData;
+import backtraceio.library.models.BacktraceResult;
+import backtraceio.library.models.types.HttpException;
+
+public class CustomRequestHandler extends URLRequestHandler {
+    CustomRequestHandler(BacktraceCredentials credentials) {
+        super(credentials);
+    }
+    private static final String LOG_TAG = CustomRequestHandler.class.getSimpleName();
+    private static final int CHUNK_SIZE = 128 * 1024;
+
+    private HttpURLConnection setupConnection(String serverUrl) throws IOException {
+        HttpURLConnection urlConnection = null;
+
+        URL url = new URL(serverUrl);
+        urlConnection = (HttpURLConnection) url.openConnection();
+        urlConnection.setRequestMethod("POST");
+        urlConnection.setUseCaches(false);
+        urlConnection.setDoOutput(true);
+        urlConnection.setDoInput(true);
+        urlConnection.setChunkedStreamingMode(CHUNK_SIZE);
+        urlConnection.setRequestProperty("Connection", "Keep-Alive");
+        urlConnection.setRequestProperty("Cache-Control", "no-cache");
+
+        urlConnection.setRequestProperty("Content-Type",
+                MultiFormRequestHelper.getContentType());
+
+        return urlConnection;
+    }
+
+    private static String getResponse(HttpURLConnection urlConnection) throws IOException {
+        InputStream inputStream;
+        if (urlConnection.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) {
+            inputStream = urlConnection.getInputStream();
+        } else {
+            inputStream = urlConnection.getErrorStream();
+        }
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+
+        StringBuilder responseSB = new StringBuilder();
+        String line;
+        while ((line = br.readLine()) != null) {
+            responseSB.append(line);
+        }
+        br.close();
+        return responseSB.toString();
+    }
+
+    private BacktraceResult getResult(HttpURLConnection urlConnection, BacktraceData data) throws IOException, HttpException {
+        BacktraceResult result;
+        int statusCode = urlConnection.getResponseCode();
+        if (statusCode == HttpURLConnection.HTTP_OK) {
+            result = BacktraceSerializeHelper.backtraceResultFromJson(
+                    getResponse(urlConnection));
+            result.setBacktraceReport(data.report);
+        } else {
+            String message = getResponse(urlConnection);
+            message = (BacktraceStringHelper.isNullOrEmpty(message)) ?
+                    urlConnection.getResponseMessage() : message;
+            throw new HttpException(statusCode, String.format("%s: %s", statusCode, message));
+        }
+        return result;
+    }
+
+    private BacktraceResult getResult(HttpURLConnection urlConnection, BacktraceNativeData data) throws IOException, HttpException {
```

Comment:
this method and 
`private BacktraceResult getResult(HttpURLConnection urlConnection, BacktraceData data) throws IOException, HttpException {` are same but only with different param, are you able to unify and reuse them?